posthog-core/src/utils.ts

```diff
@@ -114,7 +114,7 @@ export async function retriable<T>(fn: () => Promise<T>, props: RetriableOptions
   for (let i = 0; i < props.retryCount + 1; i++) {
     if (i > 0) {
       // don't wait when it's the last try
-      await new Promise((r) => setTimeout(r, props.retryDelay))
+      await new Promise<void>((r) => setTimeout(r, props.retryDelay))
```

Comment:
Need to add the explicit type here because Node expects a `Node.Duration` whereas React Native returns a `number`